README.md

```diff
@@ -93,57 +93,63 @@ dicom-anonymizer InputFilePath OutputFilePath --dictionary dictionary.json
 ```
 
 
-## Custom actions from dictionary file
+## Custom/overrides actions
 
-If you use the file anonymizer.py, then you can add custom actions:
-```json
-{
-    "(0x0002, 0x0002)": "ActionNameNotReferenced",
-    "(0x0003, 0x0003)": "ActionNameNotReferenced",
-    "(0x0004, 0x0004)": "ActionNameNotReferenced1",
-    "(0x0005, 0x0005)": "ActionNameNotReferenced2"
-}
-```
-Then, modify your file anonymizerUser.py :
+Here is a small example which keep all metadata but update the series description
+by adding a suffix passed as a parameter.
 ```python
-import dicomanonymizer.anonymizer
-
-# Functions need to have in input a dataset and a tag
-def ActionNameNotReferenced(dataset, tag):
-    # Extract the element from the dataset :
-    element = dataset.get(tag)
-    # You can update the value as here :
-    element.value = ''
-	print('custom action')
-def ActionNameNotReferenced1(dataset, tag):
-	print('custom action 1')
-def ActionNameNotReferenced2(dataset, tag):
-	print('custom action 2')
-
-actionsTagMap = {
-	(0x0001, 0x0001): "keep", #will use the default method in the dicom anonymizer
-	(0x0002, 0x0002): ActionNameNotReferenced,
-	(0x0003, 0x0003): ActionNameNotReferenced2
-}
-
-# Generate a custom dictionary that will be override the default one
-actionsMap = dicomanonymizer.anonymizer.generateActionsDictionary(actionsTagMap)
-
-# Send the map to the main function
-dicomanonymizer.anonymizer.anonymize(actionsMap)
+import argparse
+from dicomanonymizer import *
+
+def main():
+    parser = argparse.ArgumentParser(add_help=True)
+    parser.add_argument('input', help='Path to the input dicom file or input directory which contains dicom files')
+    parser.add_argument('output', help='Path to the output dicom file or output directory which will contains dicom files')
+    parser.add_argument('--suffix', action='store', help='Suffix that will be added at the end of series description')
+    args = parser.parse_args()
+
+    InputPath = args.input
+    OutputPath = args.output
+
+    dictionary = {}
+
+    def setupSeriesDescription(dataset, tag):
+        element = dataset.get(tag)
+        if element is not None:
+            element.value = element.value + '-' + args.suffix
```

Comment:
I'm not sure that's a good idea for different reasons:
- The action is always launched by calling `action(dataset, tag)`, which means that we cannot add an other parameter
- Even if we find a way to call the action method with a new attribute, how can we pass to the simpledicomanonymizer the suffix ? I mean that if we want different suffixes for different tags, how can we pass it to the dicom-anonymizer?

The core actions referenced in the dicom-anonymizer represent the standard of the dicom anonymization. Adding a suffix looks like a personal use case.

Reply:
how about a `regexp`/`rename` action ? that would be more generic that way

Reply:
Not sure to understand how to use regexp in that case.

Reply:
The `suffix` action would only add a suffix.
The suggested `regexp` action would be more generic by allowing adding suffix, prefix or anything else (something like `sed` in bash)
You would still need to be able to "configure" the action from the command line though. Is it what is giving you troubles ?

Reply:
The main issue is not the method. It's how to call it inside the dicom-anonymizer lib. Because now, we call actions with two arguments. Which is the default behavior for all standard methods. If you want to add a suffix or anything else, it means that we have to pass a third argument.

Reply:
ok to pass a third argument.

Otherwise you can use a factory of action: `actionFactory(param)(dataset, tag)` where `actionFactory()` returns an `action`